app/subscriber/src/components/content-list/ContentList.tsx

```diff
@@ -0,0 +1,92 @@
+import moment from 'moment';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Checkbox, Col, IContentModel, Row, Show } from 'tno-core';
+
+import { IGroupByState, IToggleStates } from './interfaces';
+import * as styled from './styled';
+import { determineToneIcon, groupContent, truncateTeaser } from './utils';
+
+export interface IContentListProps {
+  /** content is an array of content objects to be displayed and manipulated by the content list*/
+  content: IContentModel[];
+  /** toggleStates is an object that contains the state of the toggleable content information */
+  toggleStates: IToggleStates;
+  /** groupBy is the key by which the content will be grouped */
+  groupBy: IGroupByState;
+  /** determine the selected content based on the checkbox */
+  setSelected: React.Dispatch<React.SetStateAction<IContentModel[]>>;
+}
+
+export const ContentList: React.FC<IContentListProps> = ({
+  content,
+  toggleStates,
+  groupBy = 'source',
+  setSelected,
+}) => {
+  const navigate = useNavigate();
+  const grouped = groupContent(groupBy, content);
+
+  // State to keep track of checked items
+  const [checkedItems, setCheckedItems] = React.useState<Set<string>>(new Set());
```

Comment:
Used to help highlight the rows when they are checked, also learned that sets (although may not make a big difference here) can be much more performant than using arrays especially for use cases like this

Reply:
Reminder for self: change to `<number>` so I don't have to do an unnecessary cast